### ISSUE TITLE
feat: Azure Blob Storage attachment support

### ISSUE THREAD

**Issue by drodil** [drodil/backstage-plugin-qeta#136]
Apart from AWS S3, there could be support to upload attachments to Azure Blob Storage

**Comment by jmezach** [comment]
This could be useful for our organization and would be willing to contribute something for this. Do you have any pointers for what this might look like?

**Comment by drodil** [comment]
Basically it would require integration to blob storage (https://learn.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-nodejs) and then needed configuration values to the plugin config (auth, account name, container name etc.) with sane defaults of course. Backend changes should be placed in the attachments.ts route and the related files.

**Comment by jmezach** [comment]
That doesn't look too complicated 👍. I'll see if I can get something together.

**Comment by jmezach** [comment]
Looking at it a bit more it seems that both the GET and DELETE operations have some specific logic in `attachments.ts` for specific storage solutions, while the POST operation is entirely implemented in the corresponding Storage class. Would it make sense to refactor this a bit so that all Storage classes implement a common interface so that we could plugin alternative implementations fairly easily?

**Comment by drodil** [comment]
Yeah makes sense, it's getting quite bloated with new storage providers

**Comment by jmezach** [comment]
@drodil Created PR #255
